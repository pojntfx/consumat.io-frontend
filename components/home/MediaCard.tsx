import Link from "next/link";
import { Media } from "../../lib/api/consumat-io";
import styles from "../../styles/MediaList.module.css";
import MediaImage from "../helper/MediaImage";
import { imageSizes, useImage } from "../../hooks/ImageHook";

type MediaCardProps = {
  mediaItem: Media;
};

const MediaCard = ({ mediaItem }: MediaCardProps) => {
  return (
    <Link href={`/${mediaItem.__typename.toLowerCase()}/${mediaItem.code}`}>
      <a>
        <div
          className={
            styles.container +
            " place-self-center w-32 flex flex-col items-center justify-center rounded cursor-pointer shadow hover:shadow-md duration-75"
          }
        >
          <div className="overflow-hidden rounded">
            <MediaImage
              imageSrc={useImage(imageSizes.poster.w185, mediaItem.posterPath)}
              className={styles.img + " w-32 h-48"}
            />
          </div>

          <p
            className="bg-white z-10 text-sm py-2 px-1 text-center truncate w-full -mt-4 rounded"
            title={mediaItem.title}
          >
            {mediaItem.title}
          </p>
        </div>
      </a>
    </Link>
  );
};

export default MediaCard;
