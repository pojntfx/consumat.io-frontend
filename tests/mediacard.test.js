import React from "react";
import renderer from "react-test-renderer";
import MediaCard from "../components/home/MediaCard";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(<MediaCard mediaItem={testExamples.exampleMediaItem} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
