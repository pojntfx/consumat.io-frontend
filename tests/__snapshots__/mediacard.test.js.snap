// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/tv/117378"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="container place-self-center w-24 sm:w-32 flex flex-col items-center justify-center rounded cursor-pointer shadow hover:shadow-md duration-75"
  >
    <div
      className="overflow-hidden rounded"
    >
      <div
        className="img w-24 h-36 sm:w-32 sm:h-48 flex-shrink-0"
      >
        <img
          className="h-full w-full object-cover"
          onError={[Function]}
          src="https://image.tmdb.org/t/p/w185/l38PZlYba97ziaAsK3N3NPnQDtr.jpg"
        />
      </div>
    </div>
    <p
      className="bg-gradient-to-br text-white z-10 text-sm py-2 px-1 text-center truncate w-full -mt-4 rounded from-green-500 to-blue-500"
      title="Mouse"
    >
      Mouse
    </p>
  </div>
</a>
`;
